lib/spack/spack/test/spec_yaml.py

```diff
@@ -148,15 +148,15 @@ def test_ordered_read_not_required_for_consistent_dag_hash(
         #
         # Dict & corresponding YAML & JSON from the original spec.
         #
-        spec_dict = spec.to_dict()
-        spec_yaml = spec.to_yaml()
-        spec_json = spec.to_json()
+        spec_dict = spec.to_dict(hash=ht.full_hash)
+        spec_yaml = spec.to_yaml(hash=ht.full_hash)
+        spec_json = spec.to_json(hash=ht.full_hash)
```

Comment:
The test is called `test_ordered_read_not_required_for_consistent_dag_hash` but I think we should now call it `test_ordered_read_not_required_for_consistent_hash` and parametrize it by hash type.  So, e.g.:

```python
@pytest.mark.parametrize("hash_type", [ht.full_hash, ht.dag_hash, ht.build_hash])
test_ordered_read_not_required_for_consistent_hash(config, mock_packages, hash_type):
    ...
```

Reply:
This test compares the values by full hash, and therefore trivially fails if the yaml round-tripping is by `ht.dag_hash`. I don't think it's worth refactoring to include that (we should never write the yaml based on the DAG hash going forward).